hartsfield/api/datahub_archive_url_fetch.py

```diff
@@ -22,79 +22,72 @@
 "AS IS". REGENTS HAS NO OBLIGATION TO PROVIDE MAINTENANCE, SUPPORT, UPDATES,
 ENHANCEMENTS, OR MODIFICATIONS.
 """
-from collections import OrderedDict
-import json
-
-from flask import current_app as app
-from hartsfield import __version__ as version
-from hartsfield.configs import load_configs
-from hartsfield.api.config_controller import load_json
-from hartsfield.lib.http import tolerant_jsonify
-from hartsfield.lib.util import get_eb_environment
-
-import requests
 import datetime
+import json
+import re
 
-from google.oauth2 import service_account
+from flask import current_app as app, request
 from google.cloud import storage
+from google.oauth2 import service_account
+from hartsfield.api.auth_helper import authorzied_user_required
+from hartsfield.lib.http import tolerant_jsonify
+import hartsfield.api.read_aws_secret
 
 PUBLIC_CONFIGS = [
     'DEV_AUTH_ENABLED',
     'HARTSFIELD_ENV',
     'TIMEZONE',
 ]
 
-gcp_json_credentials = app.config['GCP_JSON_CREDENTIALS']
-gcp_json_credentials_dict = json.loads(gcp_json_credentials)
+AWS_SECRETS_NAME_GCP_JSON_CREDENTIALS = app.config['AWS_SECRETS_NAME_GCP_JSON_CREDENTIALS']
 
-# TODO: pass in gs url as input value to @app.route('/api/fetch_url_direct') from form user front-end form submission
-gs_source_url="gs://ucb-datahub-archived-homedirs/spring-2021/datahub.berkeley.edu/peterphu-2edo.tar.gz"
-# This will probably be request.args['gs_source_url'] in the def block...but that whole "request" business needs to be brought in etc.
 
-@app.route('/api/fetch_url_direct')
+gcp_json_credentials_from_aws = hartsfield.api.read_aws_secret.read_aws_secret(AWS_SECRETS_NAME_GCP_JSON_CREDENTIALS)
+gcp_json_credentials_dict = json.loads(gcp_json_credentials_from_aws)
+
+@app.route('/api/fetch_url_direct', methods=['POST'])
+@authorzied_user_required
 def fetch_url_direct():
 
-    # parse the input gs url to get bucket and blob names
-    bucket_and_blob_string = gs_source_url.replace("gs://", "")
-    bucket_and_blob_list = bucket_and_blob_string.split("/")
+    params = request.get_json()
+    gs_source_url = params.get('gs_source_url')
+    gs_source_url = gs_source_url.strip()
+    if not re.match(r'gs://.{3,}/.+', gs_source_url):
+        error_message = 'The submitted data \"' + gs_source_url + '\" is not a valid gs_source_url.'
+        v = {'response': error_message, 'status': 'error'}
+        return tolerant_jsonify(v)
```

Comment:
The above looks good. Although I suggest the following for line 58:
```
return tolerant_jsonify(
    {'message': error_message},
    500,
)
```
This will make for more standard front-end code (eg, checking HTTP status code).